attest/verifier/config/src/lib.rs

```diff
@@ -0,0 +1,449 @@
+        /// The list of hardening advisories that are known to be mitigated in
+        /// software at this enclave revision.
+        #[serde(default)]
+        mitigated_hardening_advisories: Vec<String>,
+    },
+}
+
+impl StatusVerifierConfig {
+    /// Build status verifier corresponding to ourself, and add it to a given
+    /// verifier
+    pub fn add_to_verifier(&self, verifier: &mut Verifier) {
+        match self {
+            Self::Mrenclave {
+                MRENCLAVE,
+                mitigated_config_advisories,
+                mitigated_hardening_advisories,
+            } => {
+                let mut mr_enclave_verifier = MrEnclaveVerifier::new(MrEnclave::from(*MRENCLAVE));
+                for advisory in mitigated_config_advisories.iter() {
+                    mr_enclave_verifier.allow_config_advisory(advisory);
+                }
+                for advisory in mitigated_hardening_advisories.iter() {
+                    mr_enclave_verifier.allow_hardening_advisory(advisory);
+                }
+                verifier.mr_enclave(mr_enclave_verifier);
+            }
+            Self::Mrsigner {
+                MRSIGNER,
+                product_id,
+                minimum_svn,
+                mitigated_config_advisories,
+                mitigated_hardening_advisories,
+            } => {
+                let mut mr_signer_verifier =
+                    MrSignerVerifier::new(MrSigner::from(*MRSIGNER), *product_id, *minimum_svn);
+                for advisory in mitigated_config_advisories.iter() {
+                    mr_signer_verifier.allow_config_advisory(advisory);
+                }
+                for advisory in mitigated_hardening_advisories.iter() {
+                    mr_signer_verifier.allow_hardening_advisory(advisory);
+                }
+                verifier.mr_signer(mr_signer_verifier);
+            }
+        }
+    }
+}
+
+/// Defines a json schema for a "trusted-measurements.json" file.
+/// See README.md for example.
+///
+/// The outermost string key of this is the release version number. This is not
+/// interpretted by the software, but could be added as a debug string or
+/// something, and helps with maintenance of the file.
+///
+/// The second string key, within a release, is the name of the enclave that
+/// this is a measurement for.
+///
+/// The VerifierConfig object contains measurement and hardening advisory data
+/// for that enclave at that release.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(transparent)]
+pub struct TrustedMeasurementSet {
+    table: BTreeMap<String, BTreeMap<String, StatusVerifierConfig>>,
+}
+
+impl TrustedMeasurementSet {
+    /// Create a verifier from this measurement set for a given enclave name.
+    pub fn create_verifier(&self, enclave_name: &str) -> Result<Verifier, Error> {
+        let mut count = 0usize;
+
+        let mut verifier = Verifier::default();
+        verifier.debug(DEBUG_ENCLAVE);
```

Comment:
Don't do this.

We don't support building clients that can talk to debug and non-debug instances, that's where `DEBUG_ENCLAVE` comes from.

`SGX_SIM` is used to determine which libs to link an enclave with, to be sure, but it's also used to determine whether the verifier will allow an attestations from an enclave running in debug mode, and would need to be kept in sync because every enclave does/will do node-to-node attestation. Rather than use two variables for this, we use one.

`IAS_MODE` works on conjunction with this: if `IAS_MODE == PROD || (IAS_MODE == DEV && SGX_MODE == HW)`, then we use the IAS-provided certificates as the baked-in trust root, and we talk to the relevant IAS URL (DEV or PROD)

That said, IAS DEV will never sign off on a `SGX_MODE=SW` enclave, so if you want to actually build something you can run without SGX hardware, you'll also need a mock IAS. For security reasons we generate a new, random root CA, chain, and signing key when this happens, in order to avoid shipping a cert in the repo, and also make sure that a client that doesn't verify the full integrity of an enclave (i.e. `NoDebug`/`DEBUG_ENCLAVE = false`) won't be able to connect to a real network.

Reply:
@nick-mobilecoin my thinking was, if empircally all the actual clients we build have this line in their verifiers, and it's important for making testing / releases attest correctly, i'd rather just stick it in the code here so that they do it the right way

if someone has a need to leave this part off, they can talk to us, and we can make a different API, or they can use the verifier builder correctly?

right now i think it's good that like, anything built wiht SGX_MODE = SW cannot attest to anything built with SGX_MODE = HW and vice versa, and if we leave this off, that assumption might not be true anymore

Reply:
I think that makes sense, it's probably a naming thing
```rust
verifier.debug(DEBUG_ENCLAVE)
```
This looks like it's saying enable debug enclaves